service/history/api/get_workflow_util.go

```diff
@@ -125,7 +132,14 @@ func GetOrPollMutableState(
 		}
 		defer func() { _ = eventNotifier.UnwatchHistoryEvent(workflowKey, subscriberID) }()
 		// check again in case the next event ID is updated
-		response, err = GetMutableState(ctx, shardContext, workflowKey, workflowConsistencyChecker)
+		response, err = GetMutableStateWithConsistencyCheck(
```

Comment:
I don't think this will fix the issue where error is return on L192? When error is return from there it means the cached mutable state is new enough so neither of the two new check will fail the request?

We can fix that in a later PR but would like to confirm my understanding and see if I missed something here.

Reply:
Yes. I will have it with a separate PR. And I need more time to see how I can repro the issue.